src/ram/transform/MakeIndex.cpp

```diff
@@ -274,6 +122,55 @@ Own<RamCondition> MakeIndexTransformer::constructPattern(const std::vector<std::
         }
     };
 
+    // transform condition list so that every strict inequality becomes a weak inequality + filter
+    // e.g. Tuple[level, element] < <expr> --> Tuple[level, element] <= <expr> and Tuple[level, element] !=
+    // <expr>
+    std::vector<std::unique_ptr<RamCondition>> toAppend;
+    auto it = conditionList.begin();
+    while (it != conditionList.end()) {
+        auto& cond = *it;
+        if (auto* binRelOp = dynamic_cast<RamConstraint*>(cond.get())) {
```

Comment:
Make this an early exit condition:
```
        auto* binRelOp = dynamic_cast<RamConstraint*>(it->get())
        if (binRelOp == nullptr) {
            ++it;
            continue;
        }
```
That makes it slightly more readable, and reduces the need for the big indented block.

Reply:
Great point.